gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GitFlowGraphMonitor.java

```diff
@@ -127,6 +128,14 @@ public boolean shouldPollGit() {
    */
   @Override
   void processGitConfigChanges() throws GitAPIException, IOException {
+    if (flowTemplateCatalog.isPresent() && (flowTemplateCatalog.get() instanceof ObservingFSFlowEdgeTemplateCatalog)) {
+      ObservingFSFlowEdgeTemplateCatalog catalog = (ObservingFSFlowEdgeTemplateCatalog) flowTemplateCatalog.get();
+      if (catalog.isShouldRefreshFlowGraph()) {
+        this.gitRepo.initRepository();
```

Comment:
Add a log line here indicating that we have detected updates in the template catalog and that we need to re-bootstrap the FlowGraph.